Userland/Shell/AST.cpp

```diff
@@ -2295,39 +2298,36 @@ ErrorOr<RefPtr<Value>> MatchExpr::run(RefPtr<Shell> shell)
                 if (!list[i].bytes_as_string_view().matches(pattern[i], mask_spans))
                     return false;
                 for (auto& span : mask_spans)
-                    spans.append(list[i].substring_from_byte_offset(span.start, span.length).release_value_but_fixme_should_propagate_errors());
+                    TRY(spans.try_append(TRY(list[i].substring_from_byte_offset(span.start, span.length))));
             }
 
             return true;
         }
     };
 
-    auto resolve_pattern = [&](auto& option) -> decltype(auto) {
+    auto resolve_pattern = [&](auto& option) -> Conditional<IsSame<RemoveCVReference<decltype(option)>, Regex<ECMA262>>, ErrorOr<Regex<ECMA262>>, ErrorOr<Vector<String>>> {
```

Comment:
Yuck but okay.

Reply:
I wasn't the person adding two different return types to a lambda....
I can clean it up to `ErrorOr<Conditional...>` if that is better